### PR TITLE
5 set up emailjs

### LINKED ISSUE #5: Set up : Email.js
## Summary

Set up Email.js, test it and create the function that will send the e-mails. That function will live in `src/services/contact.js` so we can import it and use it wherever needed.

### Screenshots

N/A

### Basic example

N/A

### Motivation

Users can send the team an email via the contact us page, an additional use case could be; when requesting an item from a user, we could send them an email from the dev-team as a notification.

### PR DESCRIPTION
**Pull request by OmarBassatni97**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -24,6 +24,8 @@
     "no-restricted-exports": "off",
     "react/jsx-props-no-spreading": "off",
     "import/no-cycle": "off",
-    "import/prefer-default-export": "off"
+    "import/prefer-default-export": "off",
+    "jsx-a11y/label-has-for": 0
```
Comment (Iseluin) [inline .eslintrc.json]: Big no here, we don't add eslint exceptions, we obey the rules that exists. Maybe try adding an "htmlFor" attribute to your label, that might fix the error that you will be getting.

File: src/services/contact.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const form = useRef();
+
+    const sendEmail = (e) => {
+        e.preventDefault();
+
+        emailjs.sendForm('service_iwqsu3f', 'template_d0s4lc6', form.current, 'SMrzCAgEF5PcNX7Jo')
+            .then((result) => {
+                console.log(result.text);
+            }, (error) => {
+                console.log(error.text);
```
Comment (Iseluin) [inline src/services/contact.js]: No console logs in the code please. Delete all your console logs after youre done with testing your code. You can use alerts and whatnot for the functionality.

File: src/services/contact.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { useRef } from 'react';
+import emailjs from '@emailjs/browser';
+
+export const ContactUs = () => {
```
Comment (Iseluin) [inline src/services/contact.js]: You are building a React component called ContactUs here, **what we want is the creation of the reusable JS function for sending e-mails**, not a component.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Good work, but we dont need a ContactUs React component. What we need is a reusable e-mailer function.

File: src/services/contact.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    const form = useRef();
+
+    const sendEmail = (e) => {
+        e.preventDefault();
+
+        emailjs.sendForm('service_iwqsu3f', 'template_d0s4lc6', form.current, 'SMrzCAgEF5PcNX7Jo')
```
Comment (MARWAHAKMI) [inline src/services/contact.js]: It’s recommended to save Emailjs credentials in Environment variable file.

Comment (OmarBassatni97) [reply src/services/contact.js]: Kutay said no need for that

Comment (Iseluin) [reply src/services/contact.js]: @MARWAHAKMI yeah, let's build it the way it works for now, after the project is finished we might move every credential in the env file, therefore we could get a more professional look. But for now Im not pushing anybody to do that.

Comment (OmarBassatni97) [reply src/services/contact.js]: I pushed the changes if you can check them please

Comment (Iseluin) [reply src/services/contact.js]: @OmarBassatni97 you need to ask for review again in that case, github will notify me again.

File: src/services/contact.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import emailjs from '@emailjs/browser';
+
+const contactUs = (e, templateParams) => {
```
Comment (Iseluin) [inline src/services/contact.js]: could you rename this to something relevant, like mailSender or something.

Did you test the function with a basic input you created ? Does it work properly ?

Comment (OmarBassatni97) [reply src/services/contact.js]: yes, I created a form and tested the function its works properly, and I will change the name

**Review by Iseluin — APPROVED** [review]
Great work!

**Review by sarahaboudalal — COMMENTED** [review]
Looks good and clean Omar, good job!

**Comment by MariaTheresaAzzi** [comment]
The code looks good, nice work Omar!

**Review by MariaTheresaAzzi — COMMENTED** [review]
The code looks good, nice work Omar!

**Pull request merged**